src/grasp_planner.cpp

```diff
@@ -315,6 +317,43 @@ bool GraspPlanner::computeCartesianWaypointPath(GraspCandidatePtr& grasp_candida
       return false;
     }
 
+    // Attach CO to end effector
+    if (!grasp_object_id.empty() && scene->knowsFrameTransform(grasp_object_id))
+    {
+      // Create an AttachedCollisionObject
+      moveit_msgs::AttachedCollisionObject aco;
+
+      // Create a reference to the collision object for convenience
+      moveit_msgs::CollisionObject& suction_voxel_co = aco.object;
+
+      suction_voxel_co.id = grasp_object_id;
+      suction_voxel_co.header.frame_id = ik_tip_link->getName();
+
+      // Set the aco attached link name
+      aco.link_name = ik_tip_link->getName();
+
+      // move the robot in the ps to the end of the approach path.
+      scene->setCurrentState(*start_state_copy);
+
+      // Mark object to be added
+      suction_voxel_co.operation = moveit_msgs::CollisionObject::ADD;
+      if (!scene->processAttachedCollisionObjectMsg(aco))
+      {
+        ROS_WARN_STREAM_NAMED("grasp_planner", "Failed to attach: " << aco.object.id);
+      }
+      else
+      {
+        auto start_state_with_body = std::make_shared<robot_state::RobotState>(scene->getCurrentState());
+        std::vector<const robot_state::AttachedBody*> attached_bodies;
+        start_state_with_body->getAttachedBodies(attached_bodies);
+        for (auto& ab : attached_bodies)
+          start_state_copy->attachBody(ab->getName(), ab->getShapes(), ab->getFixedTransforms(), ab->getTouchLinks(),
+                                       ab->getAttachedLinkName(), ab->getDetachPosture(), ab->getSubframeTransforms());
+        constraint_fn = boost::bind(&isGraspStateValid, scene.get(), collision_checking_verbose,
```

Comment:
maybe use std::bind or, [better yet](https://github.com/peter-can-write/cpp-notes/blob/master/prefer-lambdas-over-std::bind.md), a lambda expression?

Reply:
This simply moves the existing function